### DISCUSSION TITLE
same source different hashes in _freeze/ files from different OS machines

### DISCUSSION THREAD

**Discussion by espinielli** [quarto-dev/quarto-cli#3597] [Q&A] [answered]
Hi all,
I have set a CI workflow where Netlify published a quarto book as per [1] (we specifically use this for automatic preview on development branch [2][3]).
I am using the `freeze` option to get the code chunks pre-computed as required by the Netlify setup.
The issue is that pushes from the same source from different machine (mac and Windows) produce different hashes for the same qmd sources...and this means git merge clashes and difficulties to collaborate...
We use the same quarto version on the two OSs...

Here is an example of the clash:

```json
{
<<<<<<< HEAD
  "hash": "3575a3f572f6cd1d1124f5223cdfb566",
=======
  "hash": "ada4b65f5349f9aad31f8f739195a811",
>>>>>>> 67c67552f0aba0e2ab86d751957e6d3384296cc6
  "result": {
    "markdown": "\n# Welcome {-}\n\nThis is the website for __\"{{< meta title  >}}\"__.\n\n\nThe EUROCONTROL Business Cases team regularly refreshes the content and the underlying data\nin order to provide up-to-date information for the interested users.\n\n\n# About us\n\nEUROCONTROL Business Cases team is a dedicated team of experts supporting stakeholders to choose among different solutions and alternatives making use of the best available data. For more than 15 years, EUROCONTROL has been providing ATM decision makers with a wide range of  information enabling them to take rational long-term investment decisions.\n\n# Notice and Disclaimer {-}\n\nThe EUROCONTROL Business Cases team has made every effort to ensure that the information and analysis contained in this document are as accurate as possible. Only information from quoted sources has been used\nand information relating to named parties has been checked with the parties concerned. Despite these\nprecautions, should you find any errors or inconsistencies we would be grateful if you could please bring them to\nour attention. Our email address is [economics.business-cases@eurocontrol.int](mailto:economics.business-cases@eurocontrol.int)\n\nThis document is published by the EUROCONTROL Business Cases team in the interest of the exchange of\ninformation.\n\nInformation contained in this document may not be modified without prior written permission from the\nEUROCONTROL Business Cases team.\n\nThe views expressed herein do not necessarily respect the official views or policy of EUROCONTROL, which\nmakes no warranty, either implied or express, for the information contained in this document, neither does it\nassume any legal liability or responsibility for the accuracy, completeness or usefulness of this information.\n\n\n\n\n## Publications {-}\n\n| EUROCONTROL Headquarters\n| 96 Rue de la Fusée\n| B-1130 BRUSSELS\n| Tel: +32 (0)2 729 1152\n| Fax: +32 (0)2 729 5149\n| E-mail: publications@eurocontrol.int",
    "supporting": [],
    "filters": [
      "rmarkdown/pagebreak.lua"
    ],
    "includes": {},
    "engineDependencies": {},
    "preserve": {},
    "postProcess": true
  }
}
```

In a way it would be great to generate same hash from same source...

I initially thought this was due to line termination differences in the two OS architectures, so I forced line termination via `.gitattributes` to no avail unfortunately.

Has anybody had same/similar issues?

[1] <https://quarto.org/docs/publishing/netlify.html#rendering-on-netlify>
[2] <https://github.com/euctrl-pru/standard_inputs/pull/4>
[3] <https://deploy-preview-4--standard-inputs.netlify.app/>

**Comment by cscheid** [comment]
> I initially thought this was due to line termination differences in the two OS architectures
I think this is correct.

> so I forced line termination via .gitattributes to no avail unfortunately.
The issue, I believe, is that the hashes are being computed after loading the files, and I think there's some normalization that happens at that point. If so, this is a bug.

**Comment by ianmtaylor1** [comment]
I am experiencing this as well. I'm using quarto version 1.3.353 with Rstudio version 2023.06.1. The hash difference is resulting in the GitHub pages worker (using Ubuntu Linux) trying to re-render documents instead of using the frozen version (which was rendered using Windows 10).

**Reply by mcanouil** [reply]
Watch the following issue for updates (unfortunately, there are no updates for now):
- https://github.com/quarto-dev/quarto-cli/issues/3599

**Comment by mcanouil** [comment, marked as answer]
Closing this thread in favour of the opened issue:

- https://github.com/quarto-dev/quarto-cli/issues/3599

**Reply by cderv** [reply]
This should be now solved and available in next 1.4 (pre-release now, or soon the next release)
